spanner/row.go

```diff
@@ -378,3 +389,186 @@ func (r *Row) ToStructLenient(p interface{}) error {
 		true,
 	)
 }
+
+// SelectAll iterates all rows to the end. After iterating it closes the rows,
+// and propagates any errors that could pop up.
+// It expects that destination should be a slice. For each row it scans data and appends it to the destination slice.
+// SelectAll supports both types of slices: slice of structs by a pointer and slice of structs by value,
+// for example:
+//
+//	type Singer struct {
+//	    ID    string
+//	    Name  string
+//	}
+//
+//	var singersByPtr []*Singer
+//	var singersByValue []Singer
+//
+// Both singersByPtr and singersByValue are valid destinations for SelectAll function.
+//
+// Before starting, SelectAll resets the destination slice,
+// so if it's not empty it will overwrite all existing elements.
```

Comment:
Sorry, as mentioned before, I think we should only append to the slice (without first resetting it) in case a non-nil slice is provided. WDYT? The user may have pre-seeded the slice with some entries, or it may want to use the same slice in multiple successive calls. If we always reset, we basically preclude these options. Whereas with the alternative (just append) all these options are available (and the current behavior of resetting is trivial for the user: just pass `s[:0]` instead of just `s`)

Reply:
Removed reset logic, added unit test to validate it will append on base slice.